contracts/src/NativeTokenDestination.sol

```diff
@@ -205,7 +200,7 @@ contract NativeTokenDestination is
         lastestBurnedFeesReported = burnAddressBalance;
 
         if (reward > 0) {
-            _mint(address(this), reward);
+            _mintNativeCoin(address(this), reward);
             _deposit(reward);
```

Comment:
can we add a comment here? Took me awhile to understand why we call _mintNativeCoin, then in `_deposit()` we call `_mint` too, almost seemed like a double mint.